hierarchicalforecast/core.py

```diff
@@ -97,99 +103,142 @@ def __init__(self,
+        S_df_nw = S_df_nw.with_columns(nw.from_dict({f"{id_col}_id": np.arange(len(S_df_nw))}, 
+                                                native_namespace=self.native_namespace)[f"{id_col}_id"])
+        Y_hat_df_nw = Y_hat_df_nw.join(S_df_nw[[id_col, f"{id_col}_id"]], on=id_col, how='left')
+        Y_hat_df_nw = Y_hat_df_nw.sort(by=[f"{id_col}_id", time_col])
+        Y_hat_df_nw = Y_hat_df_nw.drop(f"{id_col}_id")
+
+        if Y_df is not None:
+            Y_df_nw = nw.from_native(Y_df)
+            Y_df_nw = Y_df_nw.join(S_df_nw[[id_col, f"{id_col}_id"]], on=id_col, how='left')
+            Y_df_nw = Y_df_nw.sort(by=[f"{id_col}_id", time_col])
+            Y_df_nw = Y_df_nw.drop(f"{id_col}_id")
+
+        S_df_nw = S_df_nw.drop(f"{id_col}_id")
 
         #----------------------------------- Check Input's Validity ----------------------------------#
+
         # Check input's validity
         if intervals_method not in ['normality', 'bootstrap', 'permbu']:
-            raise ValueError(f'Unkwon interval method: {intervals_method}')
+            raise ValueError(f'Unknown interval method: {intervals_method}')
 
         if self.insample or (intervals_method in ['bootstrap', 'permbu']):
             if Y_df is None:
-                raise Exception('you need to pass `Y_df`')
+                raise Exception('You need to provide `Y_df`.')
         
         # Protect level list
         if (level is not None):
-            level_outside_domain = np.any((np.array(level) < 0)|(np.array(level) >= 100 ))
+            level_outside_domain = np.any((np.array(level) < 0) | (np.array(level) >= 100 ))
             if level_outside_domain and (intervals_method in ['normality', 'permbu']):
-                raise Exception('Level outside domain, send `level` list in [0,100)')
+                raise ValueError("Level must be a list containing floating values in the interval [0, 100).")
 
         # Declare output names
-        drop_cols = ['ds', 'y'] if 'y' in Y_hat_df.columns else ['ds']
-        model_names = Y_hat_df.drop(columns=drop_cols, axis=1).columns.to_list()
+        model_names = list(set(Y_hat_df_nw.columns) - set([id_col, time_col, target_col]))
 
         # Ensure numeric columns
-        if not len(Y_hat_df[model_names].select_dtypes(include='number').columns) == len(Y_hat_df[model_names].columns):
-            raise Exception('`Y_hat_df`s columns contain non numeric types')
-            
-        #Ensure no null values
-        if Y_hat_df[model_names].isnull().values.any():
-            raise Exception('`Y_hat_df` contains null values')
-        
-        pi_model_names = [name for name in model_names if ('-lo' in name or '-hi' in name or '-median' in name)]
-        model_names = [name for name in model_names if name not in pi_model_names]
+        assert Y_hat_df_nw[model_names].columns == Y_hat_df_nw[model_names].select(nw.selectors.numeric()).columns, "All models in `Y_hat_df` must contain numeric values."
```

Comment:
This also looks like an expensive operation in pandas just to get the dtypes. Maybe can be done with `Y_hat_df_nw.schema` instead.
Also please use exceptions instead of assertions.